### PR TITLE
Pmmi updates, Company page

### PR DESCRIPTION
**Pull request by solocommand**
- Fix layout issue if both "products" and youtube videos are not present.
- Adds custom company page
Ref design https://sketch.cloud/s/jrga7/a/qAqdln

desktop
screencapture-0-0-0-0-9710-leaders-actuators-valves-company-13306089-bosch-rexroth-2019-10-28-17_46_24

tablet
screencapture-0-0-0-0-9710-leaders-actuators-valves-company-13306089-bosch-rexroth-2019-10-28-21_19_10

mobile
screencapture-0-0-0-0-9710-leaders-actuators-valves-company-13306089-bosch-rexroth-2019-10-28-21_19_43

### CODE REVIEW

File: packages/common/components/nodes/author-card.marko [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    alt=primaryImage.alt
+    image-modifiers=["headshot"]
+    options={ fit: "facearea", facepad: 2 }
+    width=100
+    ar="1:1"
+    link={ attrs: linkAttrs }
+  />
+  <@body>
+    <@title tag="h5" modifiers=input.titleModifiers>
+      <marko-web-obj-text type="content" obj=content field="name" tag=null link={ attrs: linkAttrs } />
```
Comment (zarathustra323) [inline packages/common/components/nodes/author-card.marko]: Use prebuilt field

File: packages/common/components/nodes/author-card.marko [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    </@title>
+    <@text modifiers=["teaser"] show=(withTitle && content.title)>
+      <marko-web-obj-text type="content" obj=content field="title" tag=null />
```
Comment (zarathustra323) [inline packages/common/components/nodes/author-card.marko]: Use prebuilt field

File: packages/common/components/nodes/author-card.marko [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    </@text>
+    <@footer>
+      <@right>
+        <if(input.withEmail && content.publicEmail)>
+          <a href=`mailto:${content.publicEmail}` class="btn btn-sm btn-primary">Email</a>
```
Comment (zarathustra323) [inline packages/common/components/nodes/author-card.marko]: Use prebuilt field

File: packages/common/components/nodes/youtube-card.marko [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { getAsObject, get } from "@base-cms/object-path";
+
+$ const video = getAsObject(input, "node");
+$ const primaryImage = getAsObject(video, "primaryImage");
+$ const withDescription = input.withDescription != null ? input.withDescription : true;
+
+$ const { linkAttrs } = input;
+
+<marko-web-node
+  type="video-content"
```
Comment (zarathustra323) [inline packages/common/components/nodes/youtube-card.marko]: This isn't used. Remove

File: packages/common/components/nodes/youtube-card.marko [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  image-position="top"
+  card=false
+  full-height=false
+  attrs=input.attrs
+>
+  $ console.log(video);
```
Comment (zarathustra323) [inline packages/common/components/nodes/youtube-card.marko]: remove console

File: packages/common/scss/common.scss [hunk 6]
```diff
@@ -41,3 +42,12 @@
     }
   }
 }
+
+.node {
+  $self: &;
+  &--contact-content-type.node--image-left {
```
Comment (zarathustra323) [inline packages/common/scss/common.scss]: Remove the node--image-left stuff

File: packages/common/scss/company-page.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
+.ldp {
+  display: flex;
+  align-items: center;
+  &__logo {
+    & .page-image {
```
Comment (zarathustra323) [inline packages/common/scss/company-page.scss]: Don't use this. Just set your own image

File: packages/common/scss/company-page.scss [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    img {
+      margin-right: 2rem;
+      border: 3px solid #fff;
+      border-radius: 100px;
+      box-shadow: inset 0 0 5px rgba(0, 0, 0, .5);
+    }
+  }
+  &__title {
+    flex-grow: 6;
```
Comment (zarathustra323) [inline packages/common/scss/company-page.scss]: Double-check what's happening here...

File: packages/common/scss/company-page.scss [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  &__social {
+    flex-grow: 1;
+  }
+  &__label {
+    display: block;
+    color: rgba(0, 0, 0, .54);
```
Comment (zarathustra323) [inline packages/common/scss/company-page.scss]: Use `$black` but confirm it's the same

File: packages/common/scss/company-page.scss [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    text-transform: uppercase;
+    &--with-hr {
```
Comment (zarathustra323) [inline packages/common/scss/company-page.scss]: `--bottom-border`

File: sites/automationworld/server/templates/content/company.marko [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+  </@head>
+  <@page>
+    <marko-web-resolve-page|{ data: content, resolved }| node=pageNode>
+      <marko-web-page-wrapper>
+        <@section>
+          <div class="row">
+            <div class="col">
+              <div class="ldp">
+                <div class="ldp__logo">
+                  <marko-web-page-image
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Do different thing here

File: sites/automationworld/server/templates/content/company.marko [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+            </@body>
+          </marko-web-node-list>
+
+          <marko-web-node-list collapsible=false class="mt-block">
+            <@header>
+              <if(content.isLeader)>Leaders in Packaging</if>
+              <else>Company Details</else>
+            </@header>
+            <@body>
+              <label class="ldp__label ldp__label--with-hr">At-a-glance</label>
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: `--bottom-border`

File: sites/automationworld/server/templates/content/company.marko [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+                    />
+                  </@slot>
+                </default-theme-card-deck-flow>
+              </@nodes>
+            </marko-web-node-list>
+          </marko-web-query>
+
+        </div>
+        <div class="col-lg-4 mb-3 mb-lg-0">
+          $ const contacts = getAsArray(content, 'contacts.edges').map(({ node }) => node);
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Quote format

File: sites/automationworld/server/templates/content/company.marko [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+              Key Contact<if(contacts.length > 1)>s</if>
+            </@header>
+            <@nodes|{ nodes }| nodes=contacts>
+              <for|node| of=nodes>
+                <common-author-card-node node=node with-email=true />
+              </for>
+            </@nodes>
+          </marko-web-node-list>
+
+          $ const videos = getAsArray(content, 'videos.edges').map(({ node }) => node);
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Quote format

File: sites/automationworld/server/templates/content/company.marko [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+            </@header>
+            <@nodes|{ nodes }| nodes=videos>
+              <for|node| of=nodes>
+                <div class="node-list__node">
+                  <common-youtube-card-node node=node />
+                </div>
+              </for>
+            </@nodes>
+            <@footer>
+              <a href=content.youtube.url target="_blank" rel="noopener" class="btn btn-sm btn-block btn-secondary">
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Deep access

File: sites/automationworld/server/templates/content/company.marko [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+                View all videos &gt;
+              </a>
+            </@footer>
+          </marko-web-node-list>
+
+          <marko-web-query|{ nodes }|
+            name="all-published-content"
+            params={ contentTypes: ["PressRelease"], limit: 3, queryFragment }
+          >
+            <div class="mt-block">
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Move mt-block to flow

File: sites/automationworld/server/templates/content/company.marko [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+                </@header>
+              </website-content-list-flow>
+            </div>
+          </marko-web-query>
+
+          <marko-web-query|{ nodes }|
+            name="all-published-content"
+            params={ contentTypes: ["News"], limit: 5, queryFragment }
+          >
+            <div class="mt-block">
```
Comment (zarathustra323) [inline sites/automationworld/server/templates/content/company.marko]: Move mt-block to flow

File: packages/common/components/nodes/youtube-card.marko (hunk 4 above)
Comment (solocommand) [reply packages/common/components/nodes/youtube-card.marko]: The `--video-content-type` modifier appends the video play icon overlaya

File: packages/common/components/elements/company-logo.marko [hunk 18]
```diff
@@ -0,0 +1,20 @@
+import { get, getAsObject } from "@base-cms/object-path";
+import { buildImgixUrl } from "@base-cms/image";
+
+$ const { content } = input;
+$ const image = getAsObject(content, "primaryImage");
+$ const blockName = "company__logo";
```
Comment (zarathustra323) [inline packages/common/components/elements/company-logo.marko]: The block name shouldn't have double underscores. Double underscores signifies a block _element_. As it is now, the image element is going to be `company__logo__image`, which is "invalid." You'll want the block name to be `company-logo` -- then the element will be `company-logo__image`

**Pull request merged**